commit 180e4ba20e1449e3f339da0da24429044a84faef
Author: Michal Hrusecky <michal.hrusecky@nic.cz>
Date:   Wed Dec 12 15:08:45 2018 +0100

    mariadb: Use more sane default path
    
    Let's set default paths in a way that makes sense.
    
    Signed-off-by: Michal Hrusecky <Michal@Hrusecky.net>

diff --git a/utils/mariadb/Makefile b/utils/mariadb/Makefile
index 93d6dfe1c..be27d7670 100644
--- a/utils/mariadb/Makefile
+++ b/utils/mariadb/Makefile
@@ -33,7 +33,7 @@ PKG_BUILD_DEPENDS:=libevent2 mariadb/host
 
 CMAKE_INSTALL:=1
 
-PLUGIN_DIR:=/usr/lib/mysql/plugin
+PLUGIN_DIR:=/usr/lib/mariadb/plugin
 
 MARIADB_COMMON_DEPENDS := \
 	+libatomic \
@@ -181,7 +181,13 @@ include $(INCLUDE_DIR)/cmake.mk
 TARGET_CFLAGS+=$(TARGET_CPPFLAGS)
 
 define Package/mariadb/install/bin
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/$(2) $(1)/usr/bin
+	if [ -f $(PKG_INSTALL_DIR)/usr/bin/$(2) ]; then\
+		$(INSTALL_DIR) $(1)/usr/bin;\
+		$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/$(2) $(1)/usr/bin;\
+	else \
+		$(INSTALL_DIR) $(1)/usr/sbin;\
+		$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/$(2) $(1)/usr/sbin;\
+	fi
 endef
 
 define Package/mariadb/description/Default
@@ -339,20 +345,11 @@ CMAKE_OPTIONS += \
 	-DDISABLE_SHARED=NO \
 	-DENABLED_PROFILING=OFF \
 	-DENABLE_STATIC_LIBS=OFF \
-	-DINSTALL_DOCDIR=share/doc/mariadb \
-	-DINSTALL_DOCREADMEDIR=share/doc/mariadb \
-	-DINSTALL_MANDIR=share/man \
-	-DINSTALL_MYSQLSHAREDIR=share/mysql \
-	-DINSTALL_MYSQLTESTDIR="" \
-	-DINSTALL_PLUGINDIR=lib/mysql/plugin \
-	-DINSTALL_SBINDIR=bin \
-	-DINSTALL_SCRIPTDIR=bin \
-	-DINSTALL_SQLBENCHDIR="" \
-	-DINSTALL_SUPPORTFILESDIR=share/mysql \
-	-DINSTALL_UNIX_ADDRDIR=/var/run/mysqld/mysqld.sock \
-	-DMYSQLD_USER=mariadb \
-	-DMYSQL_DATADIR=/var/lib/mysql \
-	-DMYSQL_UNIX_ADDR=/var/run/mysqld/mysqld.sock \
+    -DINSTALL_MYSQLSHAREDIR=share/mariadb \
+    -DINSTALL_LAYOUT=RPM \
+    -DINSTALL_PLUGINDIR=lib/mariadb/plugin \
+    -DINSTALL_MYSQLDATADIR=/srv/mysql \
+    -DINSTALL_UNIX_ADDRDIR=/var/run/mysql.sock
 	-DSKIP_TESTS=ON \
 	-DWITH_ASAN=OFF \
 	-DWITH_EMBEDDED_SERVER=OFF \
@@ -430,25 +427,23 @@ define Host/Install
 endef
 
 define Build/InstallDev
-	$(INSTALL_DIR) $(2)/bin $(1)/usr/bin $(1)/usr/include $(1)/usr/lib/mysql $(1)/usr/lib/pkgconfig $(1)/usr/share/aclocal
+	$(INSTALL_DIR) $(2)/bin $(1)/usr/bin $(1)/usr/include $(1)/usr/lib $(1)/usr/lib/pkgconfig $(1)/usr/share/aclocal
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/mysql_config $(1)/usr/bin
 	$(LN) $(STAGING_DIR)/usr/bin/mysql_config $(2)/bin
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/mysql $(1)/usr/include
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmysqlclient*.so* $(1)/usr/lib
-	cd $(1)/usr/lib/mysql; $(LN) ../libmysqlclient*.so* .
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib*/libmysqlclient*.so* $(1)/usr/lib
 	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/pkgconfig/mariadb.pc $(1)/usr/lib/pkgconfig
 	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/aclocal/mysql.m4 $(1)/usr/share/aclocal
 endef
 
 define Package/libmariadbclient/install
 	$(INSTALL_DIR) $(1)$(PLUGIN_DIR)
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libmysqlclient*.so* $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib*/libmysqlclient*.so* $(1)/usr/lib
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)$(PLUGIN_DIR)/dialog.so $(1)$(PLUGIN_DIR)
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)$(PLUGIN_DIR)/mysql_clear_password.so $(1)$(PLUGIN_DIR)
 endef
 
 define Package/mariadb-client/install
-	$(INSTALL_DIR) $(1)/usr/bin
 	$(foreach b,$(MARIADB_CLIENT),$(call Package/mariadb/install/bin,$(1),$(b));)
 	# Install convenience links for mysqlcheck multi-call binary
 	cd $(1)/usr/bin; $(LN) mysqlcheck mysqlanalyze
@@ -457,43 +452,39 @@ define Package/mariadb-client/install
 endef
 
 define Package/mariadb-client-extra/install
-	$(INSTALL_DIR) $(1)/usr/bin
 	$(foreach b,$(MARIADB_CLIENT_EXTRA),$(call Package/mariadb/install/bin,$(1),$(b));)
 endef
 
 define Package/mariadb-extra-charsets/install
-	$(INSTALL_DIR) $(1)/usr/share/mysql/charsets
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/charsets/* $(1)/usr/share/mysql/charsets
+	$(INSTALL_DIR) $(1)/usr/share/mariadb/charsets
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/charsets/* $(1)/usr/share/mariadb/charsets
 endef
 
 define Package/mariadb-server/install
-	$(INSTALL_DIR) $(1)/usr/bin
 	$(foreach b,$(MARIADB_SERVER),$(call Package/mariadb/install/bin,$(1),$(b));)
 	$(INSTALL_DIR) $(1)/etc/default
 	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_DIR) $(1)/etc/mysql/conf.d
 	$(INSTALL_BIN) files/mysqld.init $(1)/etc/init.d/mysqld
 	# DATA because spomebody thought giving CONF 0600 rights is a good idea
-	$(INSTALL_DATA) conf/my.cnf $(1)/etc/mysql
+	$(INSTALL_DATA) files/my.cnf $(1)/etc/mysql
 	$(INSTALL_DIR) $(1)$(PLUGIN_DIR)
 	# DATA because spomebody thought giving CONF 0600 rights is a good idea
 	$(INSTALL_DATA) $(PKG_INSTALL_DIR)$(PLUGIN_DIR)/daemon_example.ini $(1)$(PLUGIN_DIR)
-	$(INSTALL_DIR) $(1)/usr/share/mysql/english
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/english/errmsg.sys $(1)/usr/share/mysql/english
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/fill_help_tables.sql $(1)/usr/share/mysql
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/maria_add_gis_sp_bootstrap.sql $(1)/usr/share/mysql
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/mysql_performance_tables.sql $(1)/usr/share/mysql
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/mysql_system_tables.sql $(1)/usr/share/mysql
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mysql/mysql_system_tables_data.sql $(1)/usr/share/mysql
+	$(INSTALL_DIR) $(1)/usr/share/mariadb/english
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/english/errmsg.sys $(1)/usr/share/mariadb/english
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/fill_help_tables.sql $(1)/usr/share/mariadb
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/maria_add_gis_sp_bootstrap.sql $(1)/usr/share/mariadb
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/mysql_performance_tables.sql $(1)/usr/share/mariadb
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/mysql_system_tables.sql $(1)/usr/share/mariadb
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/share/mariadb/mysql_system_tables_data.sql $(1)/usr/share/mariadb
 endef
 
 define Package/mariadb-server-extra/install
-	$(INSTALL_DIR) $(1)/usr/bin
 	$(foreach b,$(MARIADB_SERVER_EXTRA),$(call Package/mariadb/install/bin,$(1),$(b));)
 endef
 
 define Package/mariadb-server/conffiles
-/etc/mysql/my.cnf
+/etc/my.cnf
 $(PLUGIN_DIR)/daemon_example.ini
 endef
 
